pages/detail/book/[id].js

```diff
@@ -0,0 +1,61 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Card from "@/components/Card/Card";
+import Bottom from "@/components/Bottom/Bottom";
+
+export default function DetailBookPage({ data }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const currentBook = data.find((book) => book.id == id);
```

Comment:
You are right that we always work with strict equality(===), but your solution doesn't work either. I had done it before, but it did not work, but only with (==).
Now I have the solution, because "id" is a string. "id" has to be parsed to int, of course.

```suggestion
  const currentBook = data.find((book) => book.id === parseInt(id));
```